From 987e665e71cf92d1b021d7d573a1b9733408eecf Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Sat, 2 Apr 2022 15:43:00 -0500
Subject: [PATCH] Fix for latest setuptools

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index 346e1db0..33365e5b 100644
--- a/setup.py
+++ b/setup.py
@@ -226,6 +226,7 @@ def finalize_options(self):
       url="https://github.com/symengine/symengine.py",
       python_requires='>=3.7,<4',
       zip_safe=False,
+      packages=['symengine'],
       cmdclass = cmdclass,
       classifiers=[
         'License :: OSI Approved :: MIT License',
